ivadomed/loader/utils.py

```diff
@@ -387,5 +387,6 @@ def update_metadata(metadata_src_lst, metadata_dest_lst):
             metadata_dest_lst = metadata_dest_lst + [SampleMetadata({})] * \
                                 (len(metadata_src_lst) - len(metadata_dest_lst))
         for idx in range(len(metadata_src_lst)):
-            metadata_dest_lst[idx]._update(metadata_src_lst[idx], TRANSFORM_PARAMS)
+            for i_dest in range(len(metadata_dest_lst)):
+                metadata_dest_lst[i_dest]._update(metadata_src_lst[idx], TRANSFORM_PARAMS)
```

Comment:
Not sure of something:
- if we go that way, we can remove the "if" thing just above, right?
- but still: at each iteration of idx, we are "overwritting" `metadata_dest_lst[i_dest]`. Not sure if that s what we want, but we may miss something.

What about [solution in Chapter 4](https://github.com/ivadomed/ivadomed/issues/320#issue-645561146)? I can't think on a case where it would not work.

Reply:
Oh my bad, it is outdated

Reply:
Oh, it is outdated